driver/rp2/rp2_dma.py

```diff
@@ -0,0 +1,79 @@
+import time
+import uctypes
+import struct
+import machine
+
+class DMA:
+    DMA_BASE  = 0x50000000
+
+    DMA_EN    = 0x01 << 0
+    HIGH_PRIO = 0x01 << 1
+    INCR_READ = 0x01 << 4
+    INCR_WRITE= 0x01 << 5
+    DREQ_PIO0_RX0 = 0x04 << 15
+    DREQ_SPI1_TX  = 0x12 << 15
+    DREQ_PERMANENT= 0x3F << 15
+    IRQ_QUIET = 0x01 << 21
+    BUSY      = 0x01 << 24
+
+    def __init__( self, channelNumber ):
```

Comment:
Should probably throw some exception if someone ever mistakenly tries to run this on platform other than rp2.